daemon/src/setup_contract.rs

```diff
@@ -166,15 +166,25 @@ pub async fn new(
         )?,
```

Comment:
Thank you very much for the clear write-up, it's great.

One question: will it actually spawn a separate thread even if we don't instruct `tokio::test` to be configured with multiple threads, like in the snippet below?

```rust
#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
```

Reply:
Yes, the threadpool for blocking operations is completely separate from that. See the docs for more details: https://docs.rs/tokio/latest/tokio/task/fn.spawn_blocking.html

Reply:
cool!